src/audio/igo_nr/igo_nr.c

```diff
@@ -0,0 +1,811 @@
+		break;
+	default:
+		comp_err(dev, "igo_nr_cmd_set_data() error: invalid cdata->cmd");
+		ret = -EINVAL;
+		break;
+	}
+
+	return ret;
+}
+
+static int32_t igo_nr_set_chan(struct comp_dev *dev, struct sof_ipc_ctrl_data *cdata)
+{
+	uint32_t val;
+	int32_t ch;
+	int32_t j;
+	int32_t ret = 0;
+
+	for (j = 0; j < cdata->num_elems; j++) {
+		ch = cdata->chanv[j].channel;
+		val = cdata->chanv[j].value;
+		comp_info(dev, "igo_nr_cmd_set_value(), channel = %d, value = %u", ch, val);
+		if (ch < 0 || ch >= SOF_IPC_MAX_CHANNELS) {
+			comp_err(dev, "igo_nr_cmd_set_value(), illegal channel = %d", ch);
+			return -EINVAL;
+		}
+
+		if (val)
+			igo_nr_set_chan_process(dev, ch);
+		else
+			igo_nr_set_chan_passthrough(dev, ch);
+	}
+
+	return ret;
+}
+
+static int32_t igo_nr_cmd_set_value(struct comp_dev *dev, struct sof_ipc_ctrl_data *cdata)
+{
+	int32_t ret;
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_ENUM:
+		comp_dbg(dev, "igo_nr_cmd_set_value(), SOF_CTRL_CMD_ENUM index=%d", cdata->index);
+		switch (cdata->index) {
+		case IGO_NR_ONOFF_SWITCH:
+			ret = igo_nr_set_chan(dev, cdata);
+			break;
+		default:
+			comp_err(dev, "igo_nr_cmd_set_value() error: invalid cdata->index %d",
+				 cdata->index);
+			ret = -EINVAL;
+			break;
+		}
+		break;
+	case SOF_CTRL_CMD_SWITCH:
+		comp_dbg(dev, "igo_nr_cmd_set_value(), SOF_CTRL_CMD_SWITCH, cdata->comp_id = %u",
+			 cdata->comp_id);
+		ret = igo_nr_set_chan(dev, cdata);
+		break;
+
+	default:
+		comp_err(dev, "igo_nr_cmd_set_value() error: invalid cdata->cmd");
+		ret = -EINVAL;
+		break;
+	}
+
+	return ret;
+}
+
+/* used to pass standard and bespoke commands (with data) to component */
+static int32_t igo_nr_cmd(struct comp_dev *dev,
+			  int32_t cmd,
+			  void *data,
+			  int32_t max_data_size)
+{
+	struct sof_ipc_ctrl_data *cdata = data;
+
+	comp_info(dev, "igo_nr_cmd()");
+
+	switch (cmd) {
+	case COMP_CMD_SET_DATA:
+		return igo_nr_cmd_set_data(dev, cdata);
+	case COMP_CMD_GET_DATA:
+		return igo_nr_cmd_get_data(dev, cdata, max_data_size);
+	case COMP_CMD_SET_VALUE:
+		return igo_nr_cmd_set_value(dev, cdata);
+	case COMP_CMD_GET_VALUE:
+		return igo_nr_cmd_get_value(dev, cdata);
+	default:
+		comp_err(dev, "igo_nr_cmd() error: invalid command");
+		return -EINVAL;
+	}
+}
+
+static void igo_nr_process(struct comp_dev *dev,
+			   struct comp_buffer *source,
+			   struct comp_buffer *sink,
+			   int32_t src_frames,
+			   int32_t sink_frames,
+			   uint32_t source_bytes,
+			   uint32_t sink_bytes)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	buffer_invalidate(source, source_bytes);
+
+	cd->igo_nr_func(cd, &source->stream, &sink->stream, src_frames, sink_frames);
+
+	buffer_writeback(sink, sink_bytes);
+
+	/* calc new free and available */
+	comp_update_buffer_consume(source, source_bytes);
+	comp_update_buffer_produce(sink, sink_bytes);
+}
+
+static void igo_nr_get_igo_params(struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
```

Comment:
remove new line

Reply:
done.

Reply:
But I get this warning in post-commit check. Or you are saying the new line after comp_info?
![image](https://user-images.githubusercontent.com/80802098/117097762-21003880-ad9f-11eb-8e62-48e35a867b0c.png)

Reply:
@fuyuntsuo this can be fixed incrementally as I'm going to tag v1.8-rc1 today. The newline needs to go after the variable declarations i.e. after line 601 and before 602.

Reply:
Thanks! This issue has been resolved in #4152.